src/listener.rs

```diff
@@ -61,6 +61,7 @@ fn handle_new_conn(
                     peer_cert_der: peer_cert_der.clone(),
                 };
 
+                c.bootstrap_cache.add_peer(node_info.clone());
```

Comment:
Maybe at-least allow some message exchanges (so only on read) before adding to a cache instead of adding anyone who connects to us (and maybe just to displace valid people from our bootstrap cache). This makes it even easier if done on `listen()`.